src/elm/interval.coffee

```diff
@@ -196,18 +203,33 @@ module.exports.Collapse = class Collapse extends Expression
 
       # we don't handle imprecise intervals at this time
       for a in intervalsClone
-        if a.low.isImprecise?() || a.high.isImprecise?()
+        if a.low?.isImprecise?() || a.high?.isImprecise?()
```

Comment:
Alas, you go through the trouble of determining the per width based on precision, but then any datetime w/ precision other than `ms` will die here anyway.  Oh well.  We're at least _closer_ to the goal!